### PR TITLE
Implement #20939: Methods to retrieve status about encryption and sig…

### PR DESCRIPTION
**Pull request by cweiske**
…ning keys

See https://pear.php.net/bugs/20939

### CODE REVIEW

**Comment by gauthierm** [comment]
Can you include a unit test for the new API?

File: Crypt/GPG.php [hunk 1]
```diff
@@ -1392,6 +1392,27 @@ public function handleImportKeyStatus($line, array &$result)
...
     // }}}
+
+    /**
+     * Tell if there are encryption keys registered
+     *
+     * @return boolean True if the data shall be encrypted
+     */
+    public function hasEncryptKeys()
+    {
+        return count($this->encryptKeys);
```
Comment (gauthierm) [inline Crypt/GPG.php]: I'd prefer an explicit boolean statement for the return value like `count($a) > 0` rather than relying on type coercion.

**Comment by gauthierm** [comment]
The new method can be used internally in a few spots:
 * https://github.com/pear/Crypt_GPG/blob/master/Crypt/GPG.php#L1649
 * https://github.com/pear/Crypt_GPG/blob/master/Crypt/GPG.php#L2033
 * https://github.com/pear/Crypt_GPG/blob/master/Crypt/GPG.php#L1874
 * https://github.com/pear/Crypt_GPG/blob/master/Crypt/GPG.php#L2027

**Comment by cweiske** [comment]
I've added tests, use > 0 now (was an oversight of mine) and use the methods in the places you mentioned.

**Pull request merged**